EIPS/eip-6806.md

```diff
@@ -0,0 +1,156 @@
+
+interface IERC6806 {
+    /// @notice MUST be emitted when `account` is successfully
+    ///         added to or removed from the whitelist.
+    event HoldingTimeWhitelistSet(address indexed account, bool ignoreReset);
+
+    function getHoldingInfo(
+        uint256 tokenId
+    ) external view returns (address holder, uint256 holdingTime);
+
+    function setHoldingTimeWhitelistedAddress(
+        address account,
+        bool ignoreReset
+    ) external;
+}
+```
+
+**Functions**
+
+### getHoldingInfo
+
+```
+function getHoldingInfo(uint256 tokenId) external view returns (address holder, uint256 holdingTime);
+```
+
+This function returns the current holder of the specified NFT and the length of time (in seconds) the NFT has been held by the current account.
+
+* `tokenId`: The unique identifier of the NFT.
+* Returns: A tuple containing the current holder's address and the holding time (in seconds).
+
+### setHoldingTimeWhitelistedAddress
+
+```
+  function setHoldingTimeWhitelistedAddress(address account, bool ignoreReset) external;
+```
+
+This function allows the contract owner or an authorized account to specify whether a specific transfer should be ignored for the purposes of tracking holding time.
+
+* `account`: The address that should be whitelisted for holding time reset exceptions, it could be a general address of EOA/Contract.
+* `ignoreReset`: A boolean value indicating whether the holding time reset should be ignored (`true`) or not (`false`) when transferring to or from the specified `account`.
+
+## Rationale
+
+The addition of the `getHoldingInfo` and `setHoldingTimeWhitelistedAddress` functions to an extension of the ERC-721 standard enables developers to implement NFT-based applications that require holding time information and allow for selective transfer exceptions. This extension maintains compatibility with existing ERC-721 implementations while offering additional functionality for new use cases.
+
+The `getHoldingInfo` function provides a straightforward method for retrieving the holding time and holder address of an NFT. By using seconds as the unit of time for holding duration, it ensures precision and compatibility with other time-based functions in smart contracts.
+
+The `setHoldingTimeWhitelistedAddress` function introduces flexibility in determining which transfers should affect holding time calculations. By allowing the contract owner or authorized accounts to whitelist specific addresses, transfers involving these addresses will not reset the holding time. This is particularly important for NFT-based financial transactions or special cases where holding time should not be affected by transfers.
+
+Together, these functions enhance the utility of the ERC-721 standard, enabling developers to create more sophisticated applications and experiences based on NFT holding time and offering the ability to selectively ignore holding time resets during transfers to or from whitelisted addresses. 
```

Comment:
This looks more like motivation. The rationale section should describe technical decisions made within the EIP itself, not justify the EIP as a whole.